src/Datadog.Trace/Span.cs

```diff
@@ -172,6 +173,34 @@ public Span SetTag(string key, string value)
                         Context.TraceContext.SamplingPriority = SamplingPriority.UserReject;
                     }
 
+                    break;
+                case Trace.Tags.Analytics:
+                    // value is a string and can represent a bool ("true") or a double ("0.5"),
+                    // so try to parse both.
+                    // note that "1" and "0" can parse as either types,
+                    // but they mean the same thing in this case, so it's fine.
+                    bool? boolean = value.ToBoolean();
+
+                    if (boolean == true)
+                    {
+                        // always sample
+                        SetMetric(Trace.Tags.Analytics, 1.0);
+                    }
+                    else if (boolean == false)
+                    {
+                        // never sample, remove the metric
```

Comment:
ah, so this is interesting, since setting `0` is different than not setting at all.

If the metric is not set, then we inherit the sample rate from the agent.

If the metric is set to `0` then we override any agent sample rate to be `0` (don't sample).

In this case we actually do want to set to `0` instead of `null`. The idea is that if they explicitly call `span.SetTag(Trace.Tags.Analytics, false)` they are telling us to explicitly disable analytics for that span, which would mean setting as `0` instead of `null`.

Make sense?

Reply:
Good catch, thanks! Fixed in 8e5b3bb9ff8e5e6c1a72cc04385e8d1fdd19840a.